fizzbuzz/src/main/java/fizz.buzz/number/factory/BuzzNumberFactory.java

```diff
@@ -0,0 +1,16 @@
+package fizz.buzz.number.factory;
+
+import fizz.buzz.number.BuzzNumber;
+
+public class BuzzNumberFactory extends AbstractNumberFactory {
+
+    @Override
+    public Integer getDivisor() {
+        return 5;
```

Comment:
Seeing the factories reduced like this, to just two data values, makes me think of putting them as a pair/tuple into a list and iterating over them.

You could als have a parameterized number factory, i.e. 

    new ParametrizedNumberFactory(15, FizzBuzzNumber.class)